### PR TITLE
HMS-3710 : Virtual Background Module changes added 

### PR DESCRIPTION
**Pull request by A-KVerma**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -26,6 +26,7 @@ allprojects {
         jcenter()
 
         maven { url 'https://jitpack.io' }
+//        mavenLocal()
```
Comment (PratimMallick) [inline build.gradle]: This can be removed @A-KVerma

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt [hunk 2]
```diff
@@ -157,6 +164,8 @@ class MeetingViewModel(
     .setTrackSettings(hmsTrackSettings) // SDK uses HW echo cancellation, if nothing is set in builder
     .build()
 
+  val plugin = HMSVirtualBackground(hmsSDK)
```
Comment (PratimMallick) [inline app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: Rename to `virtualBackgroundPlugin`

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt [hunk 3]
```diff
@@ -870,6 +879,38 @@ class MeetingViewModel(
...
+    imageList.add("2.jpeg")
+    imageList.add("mountains.jpg")
+    imageList.add("tree.jpg")
+
+    val randomIndex = Random.nextInt(imageList.size);
+    return getBitmapFromAsset(context!!.applicationContext,imageList[randomIndex])!!
+  }
+
+
+  fun startPlugin(context: Context?) {
```
Comment (PratimMallick) [inline app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: rename to `startVirtualBackgroundPlugin`

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt [hunk 4]
```diff
@@ -870,6 +879,38 @@ class MeetingViewModel(
...
+    Log.v(TAG, "Adding Plugin")
+    //Create Plugin, can be declared on top as global variable
```
Comment (PratimMallick) [inline app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: // First create a bitmap

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt (hunk 2 above)
Comment (PratimMallick) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt:167]: Move inside `startPlugin`

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt [hunk 5]
```diff
@@ -870,6 +879,38 @@ class MeetingViewModel(
...
+    val imageBitmap = getRandomVirtualBackgroundBitmap(context)
+    plugin.setBackground(imageBitmap)
+    // add Plugin from sdk,
+    hmsSDK.addPlugin(plugin, 15)
+  }
+
+  fun stopPlugin() {
+    //sdk.removePlugin(plugin) in try catch
+    try{
```
Comment (PratimMallick) [inline app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: Remove try catch

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt [hunk 6]
```diff
@@ -870,6 +879,38 @@ class MeetingViewModel(
...
+  }
+
+
+  fun startPlugin(context: Context?) {
+    Log.v(TAG, "Adding Plugin")
+    //Create Plugin, can be declared on top as global variable
+    val imageBitmap = getRandomVirtualBackgroundBitmap(context)
+    plugin.setBackground(imageBitmap)
+    // add Plugin from sdk,
+    hmsSDK.addPlugin(plugin, 15)
```
Comment (PratimMallick) [inline app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: Remove argument 15

Comment (PratimMallick) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: Add HMSResultActionListener

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt (hunk 2 above)
Comment (A-KVerma) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: done

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt (hunk 3 above)
Comment (A-KVerma) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: done

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt (hunk 4 above)
Comment (A-KVerma) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: done

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt (hunk 6 above)
Comment (A-KVerma) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: done

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt (hunk 5 above)
Comment (A-KVerma) [reply app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt]: done

**Review by PratimMallick — APPROVED** [review]

**Review by PratimMallick — APPROVED** [review]

**Pull request merged**